docker/lib/dependabot/docker/update_checker.rb

```diff
@@ -146,8 +146,7 @@ def fetch_latest_version(version)
           # Prune out any downgrade tags before checking for pre-releases
           # (which requires a call to the registry for each tag, so can be slow)
           candidate_tags = comparable_tags_from_registry(version)
-          non_downgrade_tags = remove_version_downgrades(candidate_tags, version)
-          candidate_tags = non_downgrade_tags if non_downgrade_tags.any?
+          candidate_tags = remove_version_downgrades(candidate_tags, version)
```

Comment:
❓ None of the tests fail when I removed the logic to use _all_ `candidate_tags` if the `non_downgrade_tags` fail, is there a reason we would want to fail over to using all tags I'm not seeing?

Do we have a missing test case?

Reply:
I can imagine there could be some tricky tags that might trip this up but it would be better to identify & handle those rather than let it continue without filtering?